### PR TITLE
SQL Array docs

### PR DESCRIPTION
**Pull request by juggernaut**
As promised in https://github.com/jdbi/jdbi/issues/1261 , I've written up some basic docs for the Sql Arrays section based on the TODOs.

### CODE REVIEW

**Comment by qualidafial** [comment]
A few details we could add that would make this better:

- SqlArrayType is used for binding arrays. You can map arrays of any type with a registered column mapper.
- You can bind Java arrays or collections to array columns
- Array columns can be mapped to any container type registered with JdbiCollectors. E.g. a `VARCHAR[]` may be mapped into an `ImmutableList<String>` if the guava plugin is installed.

**Comment by juggernaut** [comment]
Thanks for the feedback! I've added an example that shows binding collections instead of arrays, and another section for mapping custom types. Let me know if this looks good.

File: docs/src/adoc/index.adoc [hunk 1]
```diff
@@ -1608,6 +1608,22 @@ int[] userIds = handle.createQuery("select user_ids from groups where id = :id")
...
 ----
 
+You can also use Collections in place of arrays, but you'll need to provide
+generic type information if you're using the fluent API:
+
+[source,java]
+----
+handle.createUpdate("insert into groups (id, user_ids) values (:id, :userIds)")
+      .bind("id", 1)
+      .bindByType("userIds", Arrays.asList(10, 5, 70), new GenericType<List<Integer>>() {})
```
Comment (qualidafial) [inline docs/src/adoc/index.adoc]: I was thinking we could add a `bindArray(column, collection, elementType)` method to make array binding simpler in core.

(in a separate PR)

File: docs/src/adoc/index.adoc [hunk 2]
```diff
@@ -1680,6 +1696,38 @@ handle.createUpdate("insert into groups (id, user_ids) values (:id, :users)")
...
 
+==== Mapping custom array types
+
+`SqlArrayType` only allows you to bind Java array/collection arguments to their
+SQL counterparts. To map SQL array columns back to Java types, you can register
+a regular `ColumnMapper`:
+
+[source,java]
+----
+public class UserIdListColumnMapper implements ColumnMapper<List<UserId>> {
```
Comment (qualidafial) [inline docs/src/adoc/index.adoc]: Actually, you only need to register a `ColumnMapper<UserId>` and Jdbi figures out the rest, for any type of collection that's registered with `JdbiCollectors` config class.

File: docs/src/adoc/index.adoc [hunk 3]
```diff
@@ -1680,6 +1696,38 @@ handle.createUpdate("insert into groups (id, user_ids) values (:id, :users)")
...
+                .collect(Collectors.toList());
+    }
+}
+----
+
+[source,java]
+----
+List<UserId> userIds = handle.createQuery("select user_ids from groups where id = :id")
+      .bind("id", 1)
+      .map(new UserIdListColumnMapper())
```
Comment (qualidafial) [inline docs/src/adoc/index.adoc]: This would change to e.g. `mapTo(new GenericType<List<UserId>>() {})`

File: docs/src/adoc/index.adoc (hunk 2 above)
Comment (juggernaut) [reply docs/src/adoc/index.adoc]: thanks, fixed. Much clearer now

File: docs/src/adoc/index.adoc (hunk 1 above)
Comment (juggernaut) [reply docs/src/adoc/index.adoc]: Yes, that would indeed be nice - I've created an issue for this https://github.com/jdbi/jdbi/issues/1274

**Comment by qualidafial** [comment]
@juggernaut thanks for contributing!

**Pull request merged**